script/deploy

```diff
@@ -0,0 +1,62 @@
+#!/usr/bin/env bash
+
+set -e
+
+DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
+
+## Validations
+[ -z "${DOCKER_USER}" ] && { echo "Need to set DOCKER_USER"; exit 0; }
```

Comment:
I _guess_ this could be `exit 1`, but this fails the entire CI build, not just the deploy portion. To that end, anywhere there is `exit 0` is a condition that isn't fatal to the deploy, but also is non-destructive. This becomes more a declaration of "ignore the build step" from Travis' perspective.

The only scenario where I saw a true `exit 1` is with trying to force push a new tag. My take is that once we have an artifact in the wild, that's it. It's super easy to override a tag, so this is our check against that. Otherwise, the scenarios of...

* Missing variables - for whatever reason, variables aren't set... so, just ignore the deploy.
* In a PR, meh. Ignore the deploy.